src/overlays/actors/ovl_Obj_Y2shutter/z_obj_y2shutter.c

```diff
@@ -28,24 +41,157 @@ const ActorInit Obj_Y2shutter_InitVars = {
     (ActorFunc)ObjY2shutter_Draw,
 };
 
-// static InitChainEntry sInitChain[] = {
-static InitChainEntry D_80B9AEB0[] = {
+ShutterInfo sShutterInfo[] = {
+    // rectangular shutter door, activated by floor plate
+    {
+        object_kaizoku_obj_DL_0032A0,
+        &object_kaizoku_obj_Colheader_0035B0,
+        120.0f,
+        20.0f,
+        3.0f,
+        -20.0f,
+        3.0f,
+        4,
+        8,
+        160
+    },
+    // hexagonal shutter, activated by hitting crystal switch
+    {
+        object_kaizoku_obj_DL_005720,
+        &object_kaizoku_obj_Colheader_005EC8,
+        150.0f,
+        1.0f,
+        0.04f,
+        -1.0f,
+        0.04f,
+        6,
+        0xC,
+        160
+    },
+};
+
+static InitChainEntry sInitChain[] = {
     ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneForward, 4000, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneScale, 800, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneDownward, 800, ICHAIN_STOP),
 };
 
-#endif
+void ObjY2shutter_Init(Actor* thisx, PlayState* play) {
+    s32 pad[2];
+    ShutterInfo* info = &sShutterInfo[OBJY2SHUTTER_GET_SHUTTER_TYPE(thisx)];
+    ObjY2shutter* this = THIS;
+    
+    Actor_ProcessInitChain(&this->dyna.actor, sInitChain);
+    DynaPolyActor_Init(&this->dyna, 0);
+    DynaPolyActor_LoadMesh(play, &this->dyna, info->colHeader);
+}
+
+void ObjY2shutter_Destroy(Actor* thisx, PlayState* play) {
+    ObjY2shutter* this = THIS;
+
+    DynaPoly_DeleteBgActor(play, &play->colCtx.dyna, this->dyna.bgId);
+}
+
+void func_80B9AA20(ObjY2shutter* this, ShutterInfo* info, s32 shutterType) {
+    this->unk_15C = info->unk_1E;
+    this->unk_15F = info->unk_1C;
+    if (shutterType == 0) {
+        Actor_PlaySfxAtPos(&this->dyna.actor, NA_SE_EV_METALDOOR_OPEN);
+    }
+}
 
-extern InitChainEntry D_80B9AEB0[];
+void ObjY2shutter_Update(Actor* thisx, PlayState* play) {
+    s32 pad;
+    ObjY2shutter* this = THIS;
+    s32 shutterType = OBJY2SHUTTER_GET_SHUTTER_TYPE(&this->dyna.actor);
+    ShutterInfo* info = &sShutterInfo[shutterType];
+    f32 sp2C = this->dyna.actor.world.pos.y;
+    f32 sp28 = 0.0f;
+    f32 sp24 = 0.0f;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Y2shutter/ObjY2shutter_Init.s")
+    sShutterInfo[0].unk_1E = gGameInfo->data[0x2F4] + 0xA0;
+    sShutterInfo[1].unk_1E = gGameInfo->data[0x2F5] + 0xA0;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Y2shutter/ObjY2shutter_Destroy.s")
+    if (((shutterType == 0) && (gGameInfo->data[0x2F6] != 0)) ||
+        ((shutterType != 0) && (gGameInfo->data[0x2F7] != 0))) {
```

Comment:
The gGameInfo stuff is written using the REG macros, you can use regconvert.py to do this (or do it manually).